tools/cuds-prioritized-attribution/composer/main.py

```diff
@@ -0,0 +1,225 @@
+
+        Args:
+            bq_client: bigquery.Client object
+            env_vars: Dictionary of string key-value pairs of environment vars.
+
+        Returns:
+             None
+        """
+        execute_query(bq_client,
+                      env_vars,
+                      distribute_commitment,
+                      env_vars['distribute_commitments_table'],
+                      False)
+
+    def billing_output(bq_client: bigquery.Client,
+        env_vars: Dict[str, Union[str, bool]]) -> None:
+        """Executes third query to format output schema.
+
+        Args:
+            bq_client: bigquery.Client object
+            env_vars: Dictionary of string key-value pairs of environment vars.
+
+        Returns:
+             None
+        """
+        execute_query(bq_client,
+                      env_vars,
+                      billingoutput,
+                      env_vars['corrected_table_name'],
+                      True)
+
+    def delete_temp_tables(bq_client: bigquery.Client,
+        env_vars: Dict[str, Union[str, bool]]) -> None:
+        """Deletes the three temporary tables that were created by the DAG.
+
+        Args:
+            bq_client: bigquery.Client object
+            env_vars: Dictionary holding key-value pair of environment vars.
+
+        Returns:
+            None
+        """
+        helper_function.delete_table(bq_client,
+                                     env_vars['corrected_dataset_id'],
+                                     env_vars['temp_commitments_table_name'])
+        helper_function.delete_table(bq_client,
+                                     env_vars['corrected_dataset_id'],
+                                     env_vars['distribute_commitments_table'])
+        helper_function.delete_table(bq_client,
+                                     env_vars['corrected_dataset_id'],
+                                     env_vars['project_label_credit_breakout_table'])
+
+    # Obtain values for all of the environment variables
+    KEY_LIST = ['project_id', 'billing_export_table_name',
+                'corrected_dataset_id', 'corrected_table_name',
+                'commitments_table_name', 'enable_cud_cost_attribution', 'cud_cost_attribution_option']
+    ENV_VARS = get_env_variables(KEY_LIST)
+    # Create temp tables for each of the three queries
+    ENV_VARS['distribute_commitments_table'] = 'temp_distribute_commitments_table'
+    ENV_VARS['project_label_credit_breakout_table'] = 'temp_project_label_credit_data_table'
+    ENV_VARS['temp_commitments_table_name'] = 'temp_commitments_table'
+    # Convert string to bool because environment variables are strings.
+    ENV_VARS['enable_cud_cost_attribution'] = (
+        ENV_VARS['enable_cud_cost_attribution'].lower() == 'true'
+    )
+    ENV_VARS['cud_cost_attribution_option'] = 'b' if ENV_VARS['cud_cost_attribution_option'].lower() == 'b' else 'a'
+    bq_client = bigquery.Client()
```

Comment:
This is super not kosher to define client in dag definition file airflow.
We should be leveraging the bigquery hook to handle authentication of the workers, etc.
Is bq client json serializable?
Will defining this client in the dag definition file and passing it to python operator even work on distributed workers?

Reply:
@kardiff18 fixed it now.